.github/pull_request_template.md

```diff
@@ -0,0 +1,21 @@
+This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md).
+
+# Author
+<!-- NOTE: Do not modify these when initially opening the pull request.  This is a checklist template that you tick off AFTER the pull request is created. -->
+- [ ] Self-review: Did you review your own code in GitHub's web interface? Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs.
+- [ ] Automated tests: Did you add appropriate automated tests for any code changes?
+- [ ] Code coverage: Did you check the code coverage report for the automated tests? Coverage can be generated by running `./gradlew check -PisCoverageEnabled=true; ./gradlew jacocoTestReport -PisCoverageEnabled=true` or by downloading the test report from the CI results. While we are not looking for perfect coverage, the tool can point out potential cases that have been missed.
```

Comment:
I think we can remove the coverage thing until we settle on how to we want it to be. Have you worked with coverage and SwiftUI? how does it affect metrics ?

Reply:
Ideally you would not unit test your UI layer, and instead, remove any business logic from the view itself and put it in a more appropriate ViewModel or something where it can be unit tested.  The more you can keep your UI layer as simply a presentation layer the better.  There will be some logic in your views, like conditionally showing one view or another, and this could eventually be Snapshot tested, but to start I would suggest focusing on moving as much logic as you can into a layer that can actually be unit tested well and worry about the UI portion at a later date when or if it is needed.

Reply:
This step should just be about getting authors to run the coverage tool and look at the output.  So this could be less about a metric and more about encouraging use of a tool to make important test omissions obvious.

It will usually make it very obvious if an important method was missed in terms of coverage, and then the author will be motivated to improve it.

Should using automation to block merging be a different discussion?  For example, to @adamstener 's points out that view code might have less coverage.  That would negatively impact metrics, but perhaps the metric isn't the thing we want to optimize for.

Reply:
I would imagine that any code coverage automation you have could ignore files or folders of UI code so that metrics aren't skewed.

Reply:
Until we have a tool for this though, I agree with @pacu that it should be removed, it will probably only add confusion at this point to reference a tooling for android that is not present in this repo.

Reply:
I think we just need to tweak the instructions to include something like this: https://www.swiftbysundell.com/tips/gathering-test-coverage-in-xcode/

Reply:
I'm adding a doc section to show how to check the code coverage %